specification/resourcegraph/resource-manager/Microsoft.ResourceGraph/preview/2018-09-01-preview/graphquery.json

```diff
@@ -0,0 +1,535 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure Resource Graph Query",
+    "description": "Azure Resource Graph Query API Reference",
+    "version": "2018-09-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroup/{resourceGroupName}/providers/Microsoft.ResourceGraph/queries": {
+      "get": {
+        "description": "Get all graph queries defined within a specified subscription and resource group.",
+        "operationId": "GraphQuery_List",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "A list containing 0 or more graph queries.",
+            "schema": {
+              "$ref": "#/definitions/GraphQueryListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/GraphQueryError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
```

Comment:
>null [](start = 26, length = 4)

If you ever want to introduce paging in the future, you would have to introduce a new API version and you'll have constrain the number of results you return on the old API. My suggestion would be to set this to "nextLink" even if you are returning one page currently. That way you you can introduce paging later without breaking clients.

Reply:
+1

---
In reply to: [278377348](https://github.com/Azure/azure-rest-api-specs/pull/5478#discussion_r278377348) [](ancestors = 278377348)

Reply:
NextLink is still null here.

---
In reply to: [278412814](https://github.com/Azure/azure-rest-api-specs/pull/5478#discussion_r278412814) [](ancestors = 278412814,278377348)

Reply:
Just updated as well.

Reply:
See it now. Thanks!

---
In reply to: [279135228](https://github.com/Azure/azure-rest-api-specs/pull/5478#discussion_r279135228) [](ancestors = 279135228,278412814,278377348)